src/chainstate/stacks/db/blocks.rs

```diff
@@ -3836,6 +3912,18 @@ impl StacksChainState {
                        last_microblock_hash, last_microblock_seq, block.block_hash(), block.header.parent_microblock, block.header.parent_microblock_sequence);
             }
 
+            // get the burnchain block that precedes this block's sortition
+            let parent_burn_hash = SortitionDB::get_block_snapshot_consensus(
+                &burn_dbconn.tx(),
+                &chain_tip_consensus_hash,
+            )?
+            .expect(
+                "BUG: Failed to load snapshot for block snapshot during Stacks block processing",
+            )
+            .parent_burn_header_hash;
+            let stacking_burn_ops =
+                SortitionDB::get_stack_stx_ops(&burn_dbconn.tx(), &parent_burn_hash)?;
```

Comment:
Do we have an upper bound on how many operations this can be?  Or do we still need to figure that out?

Reply:
There's two bounds -- one is the bitcoin block size, and the other is the compute limit. The bitcoin block size will limit this far more than the compute limit.

Reply:
:+1: Will still want to measure it when implementing the block-fill logic, but this is still a good outcome.